### PR TITLE
add support for multi-stage build in getting-started example

### LINKED ISSUE #393: Multi-stage builds support in getting-started Dockerfile
### Expected behavior
Use multi-stage builds for getting-started example. 

### Information
Currently the Dockerfile used in the getting started examples is in the "single-stage" format. As an optimisation, multi-stage format could be used

### PR DESCRIPTION
**Pull request by zanetworker**
Add multi-stage build format for getting-started Dockerfile, which resolves #393. 

Signed-off-by: zanetworker <adel.zalok.89@gmail.com>

### CODE REVIEW

File: examples/getting-started/Dockerfile [hunk 1]
```diff
@@ -1,6 +1,9 @@
-FROM golang:1.9.4-alpine3.7
-
+FROM golang:1.9.4-alpine3.7 as builder
 WORKDIR /go/src/github.com/GoogleCloudPlatform/skaffold/examples/getting-started
-CMD ["./app"]
 COPY main.go .
-RUN go build -o app main.go
+RUN CGO_ENABLED=0 GOOS=linux go build -a -installsuffix cgo -o app .
```
Comment (dgageot) [inline examples/getting-started/Dockerfile]: Do we really need `CGO_ENABLED=0 GOOS=linux`?

File: examples/getting-started/Dockerfile [hunk 2]
```diff
@@ -1,6 +1,9 @@
...
+
+FROM alpine:latest  
+WORKDIR /root/
+COPY --from=builder /go/src/github.com/GoogleCloudPlatform/skaffold/examples/getting-started/app .
+CMD ["./app"]
```
Comment (dgageot) [inline examples/getting-started/Dockerfile]: Can you move `COPY` to the end to maximize caching?

File: examples/getting-started/Dockerfile [hunk 3]
```diff
@@ -1,6 +1,9 @@
...
-
+FROM golang:1.9.4-alpine3.7 as builder
 WORKDIR /go/src/github.com/GoogleCloudPlatform/skaffold/examples/getting-started
-CMD ["./app"]
 COPY main.go .
-RUN go build -o app main.go
+RUN CGO_ENABLED=0 GOOS=linux go build -a -installsuffix cgo -o app .
+
+FROM alpine:latest  
+WORKDIR /root/
```
Comment (dgageot) [inline examples/getting-started/Dockerfile]: Is `WORKDIR` needed?

File: examples/getting-started/Dockerfile (hunk 2 above)
Comment (zanetworker) [reply examples/getting-started/Dockerfile]: > Preview

Yes that makes sense, will do that.

File: examples/getting-started/Dockerfile (hunk 3 above)
Comment (zanetworker) [reply examples/getting-started/Dockerfile]: Not really in this case, just for convenience when debugging. Removed it.

File: examples/getting-started/Dockerfile (hunk 1 above)
Comment (zanetworker) [reply examples/getting-started/Dockerfile]: No, removed it.

**Comment by zanetworker** [comment]
Thanks for your review, applied the changes.

**Pull request merged**